packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -542,77 +542,11 @@ function* computeObjectNodeDiscrepancies(
 	}
 }
 
-/**
- * This function uses incompatibilities to determine if changes to a document schema are backward-compatible, i.e., it determines
- * whether the `view` schema allows a superset of the documents that the `stored` schema allows.
- *
- * @remarks
- * According to the policy of schema evolution, `isRepoSuperset` supports three types of changes:
- * 1. Adding an optional field to an object node.
- * 2. Expanding the set of allowed types for a field.
- * 3. Relaxing a field kind to a more general field kind.
- *
- * Notes: We expect isRepoSuperset to return consistent results with allowsRepoSuperset. However, currently there are some scenarios
- * where the inconsistency will occur:
- * - Different Node Kinds: If view and stored have different node kinds (e.g., view is an objectNodeSchema and stored is a mapNodeSchema),
- * `isRepoSuperset` will determine that view can never be the superset of stored. In contrast, `allowsRepoSuperset` will continue
- * validating internal fields.
- *
- * TODO: Evaluate if this function is needed at all. It is only used in tests and could possibly be replaced with `allowsRepoSuperset`.
- */
-export function isViewSupersetOfStored(view: FieldSchema, stored: TreeStoredSchema): boolean {
-	const discrepancies = getAllowedContentDiscrepancies(view, stored);
-
-	for (const discrepancy of discrepancies) {
-		switch (discrepancy.mismatch) {
-			case "nodeKind": {
-				if (discrepancy.stored !== undefined) {
-					// It's fine for the view schema to know of a node type that the stored schema doesn't know about.
-					return false;
-				}
-				break;
-			}
-			case "valueSchema":
-			case "allowedTypes":
-			case "fieldKind": {
-				if (!isFieldDiscrepancyCompatible(discrepancy)) {
-					return false;
-				}
-				break;
-			}
-			case "fields": {
-				if (
-					discrepancy.differences.some(
-						(difference) => !isFieldDiscrepancyCompatible(difference),
-					)
-				) {
-					return false;
-				}
-				break;
-			}
-			// No default
-		}
-	}
-	return true;
-}
-
-function isFieldDiscrepancyCompatible(discrepancy: FieldDiscrepancy): boolean {
-	switch (discrepancy.mismatch) {
-		case "allowedTypes": {
-			// Since we only track the symmetric difference between the allowed types in the view and
-			// stored schemas, it's sufficient to check if any extra allowed types still exist in the
-			// stored schema.
-			return discrepancy.stored.length === 0;
-		}
-		case "fieldKind": {
-			return posetLte(discrepancy.stored, discrepancy.view, fieldRealizer);
-		}
-		case "valueSchema": {
-			return false;
-		}
-		// No default
-	}
-	return false;
+export function posetLte<T>(a: T, b: T, realizer: Realizer<T>): boolean {
```

Comment:
Looks like this is only in one place, and that's in a test, so I think it should have moved to the test code along with the rest. I believe the same is true for comparePosetElements, and likely a bit more of the code in that area.

Do that and I won't complain about these functions not being documented ;)

Reply:
comparePosetElements is used by SchemaCompatibilityTester, I'll move posetLte and add a TODO

Reply:
oops, I think I saw Tester in the path and though it was a test file.